server/api/chat/chat.controllers.js

```diff
@@ -231,17 +231,25 @@ export const getUserChatsList = asyncHandler(async (req, res, next) => {
       return next(new Error("failed retrieving"));
     }
     if (userChats.length > 0) {
-      userChats = userChats.map((chat) => {
-        const { users, messagesHistory } = chat;
-        const recieverName = users.filter((user) => user !== name)[0];
-        const lastMessage = newestMessage(messagesHistory);
-        return {
-          lastMessage: lastMessage,
-          recieverName: recieverName,
-        };
-      });
+      userChats = await Promise.all(
+        userChats.map(async (chat) => {
+          const { users, messagesHistory } = chat;
+          const recieverName = users.filter((user) => user !== name)[0];
+          const user = await User.findOne({ name: recieverName }).lean();
+          const userAvatar = user ? user.avatar : null;
+          const lastMessage = newestMessage(
+            messagesHistory,
+            name,
+            recieverName
+          );
+          return {
+            profile: userAvatar,
```

Comment:
is the keys change ok with taufiq?

Reply:
asked tawfiq its all okay , just check chat.utils should the function return 
return messagesArr[0].sender === name
    ? messagesArr[0].contentOriginal
    : messagesArr[0].contentTranslated;
    
    or return messagesArr[0].content